### PR TITLE
Spelling: Wi-Fi, turn on/off, on/off

### PR DESCRIPTION
**Pull request by comradekingu**
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/5746)

### CODE REVIEW

File: frontend/apps/reader/modules/readergesture.lua [hunk 1]
```diff
@@ -75,9 +75,9 @@ local action_strings = {
     toggle_gsensor = _("Toggle accelerometer"),
     toggle_rotation = _("Toggle rotation"),
 
-    wifi_on = _("Enable wifi"),
-    wifi_off = _("Disable wifi"),
-    toggle_wifi = _("Toggle wifi"),
+    wifi_on = _("Turn on Wi-Fi"),
+    wifi_off = _("Tun off Wi-Fi"),
```
Comment (Frenzie) [inline frontend/apps/reader/modules/readergesture.lua]: It would be preferable to do the reverse. That is, change every accidental occurrence of "Wi-Fi" to wifi.

(Also, major typo. ;-) )

File: frontend/apps/reader/modules/readergesture.lua [hunk 2]
```diff
@@ -248,7 +248,7 @@ function ReaderGesture:init()
         local reset = false
         for k, v in pairs(mirrored_if_rtl) do
             -- We only replace them if they are still the other direction's default.
-            -- If not, the user has changed them: let him deal with setting new ones if needed.
+            -- If not, the user has changed them: Let it deal with setting new ones if needed.
```
Comment (Frenzie) [inline frontend/apps/reader/modules/readergesture.lua]:
```suggestion
            -- If the user has changed directions, they can deal with setting new ones if needed.
```

They is a gender-neutral third person pronoun, but it is not normally used for people.

File: frontend/apps/reader/modules/readergesture.lua [hunk 3]
```diff
@@ -259,7 +259,7 @@ function ReaderGesture:init()
             end
         end
         if reset then
-            logger.info("UI language direction changed: resetting some gestures to direction default")
+            logger.info("UI language direction changed: Resetting some gestures to default direction.")
```
Comment (Frenzie) [inline frontend/apps/reader/modules/readergesture.lua]: While I don't typically police comments too much, our house style doesn't capitalize after colons.
```suggestion
            logger.info("UI language direction changed: resetting some gestures to default direction.")
```
```suggestion
            logger.info("UI language direction changed. Resetting some gestures to default direction.")
```

File: frontend/apps/reader/modules/readergesture.lua [hunk 4]
```diff
@@ -1601,9 +1601,9 @@ end
 function ReaderGesture:onGSensorToggle()
     local new_text
     if G_reader_settings:isTrue("input_ignore_gsensor") then
-        new_text = _("Accelerometer rotation events will now be ignored.")
+        new_text = _("Accelerometer rotation events on.")
```
Comment (Frenzie) [inline frontend/apps/reader/modules/readergesture.lua]: I'm pretty sure that's the reverse. :-P

File: frontend/apps/reader/modules/readergesture.lua (hunk 2 above)
Comment (poire-z) [reply frontend/apps/reader/modules/readergesture.lua]: If we need to be that excessively politically correct in comments, I'm gonna stop putting comments :|

File: frontend/apps/reader/modules/readergesture.lua (hunk 3 above)
Comment (poire-z) [reply frontend/apps/reader/modules/readergesture.lua]: No, it should stay `resetting some gestures to direction default`.
We're not resetting gestures to any "direction" (so, not to "default direction).
We're resetting gestures to the gestures that are the default for that new direction.

**Comment by poire-z** [comment]
Mhhh, these changes feel all gratuitous, less precise, or wrong. (I seem to prefer our current enable/disable vs turn on/off - but english being not my first language...)

**Comment by pazos** [comment]
> Mhhh, these changes feel all gratuitous, less precise, or wrong. (I seem to prefer our current enable/disable vs turn on/off - but english being not my first language...)

On linux NetworkManager is using "enable/disable" toggles for wireless and wired networks. I would stay with these too. 

For multigesture "turn on/off" doesn't make any sense as we're not turning out anything.

File: frontend/apps/reader/modules/readergesture.lua (hunk 2 above)
Comment (Frenzie) [reply frontend/apps/reader/modules/readergesture.lua]: Ftr, I'm objecting exclusively to "it".

**Comment by Frenzie** [comment]
Multigestures (or basically anything that can be toggled) are either turned on/enabled or turned off/disabled. Turn on is arguably more natural. It's an unnecessary change, but not a bad one.

**Comment by poire-z** [comment]
For things with a state, like light or wifi, I'd be ok with _turn on/off_.
For something that is a feature, that is there and can apply to stuff that needs it, or not if there's no such stuff, _turn on/off_ sounds strange to me.
For example, with EPUB rendering, I _enable floats_, I don't _turn on floats_. But I may _turn on floats support_.
Same way, I think we _enable multigestures_. Or _turn on multigestures support_. But the former is simpler, shorter, obvious :)

File: frontend/apps/reader/modules/readergesture.lua (hunk 2 above)
Comment (NiLuJe) [reply frontend/apps/reader/modules/readergesture.lua]: Yep, purely grammatical: *it* for a person just doesn't work ;).

File: frontend/apps/reader/modules/readergesture.lua (hunk 1 above)
Comment (NiLuJe) [reply frontend/apps/reader/modules/readergesture.lua]: But I kind of *like* Wi-Fi! ;p.

(In fact, most of those occurrences are probably from me ;p).

Comment (Frenzie) [reply frontend/apps/reader/modules/readergesture.lua]: I'm likely responsible for a decent amount of wifi, but it wasn't a mistake. I'm well aware of what the Wi-Fi Alliance calls itself, but in my usage hifi and wifi are fundamentally the same. That applies regardless whether we're talking about it as an abbreviation of a phrase like wireless fidelity or as a generic concept of wireless network, free from the confines of any particular trademark, patent, or technical standard.

(I recognize that in well-edited English prose, wifi is still in a minority. This will change.)

Comment (comradekingu) [reply frontend/apps/reader/modules/readergesture.lua]: If you read the actual claims of the Wi-Fi Alliance, it only concerns selling a physical product that makes use of Wi-Fi technology. By any stretch of trademark law, other uses are unenforceable, and especially now that "Wi-Fi" is a word. As it will remain being outside of whatever possibly consistent action is taken in KOReader. "wifi" just looks wrong, and pollutes the otherwise "Wi-Fi" translation memory of Weblate. Alternatives are "WLAN" and "wireless network".

File: frontend/apps/reader/modules/readergesture.lua (hunk 2 above)
Comment (comradekingu) [reply frontend/apps/reader/modules/readergesture.lua]: It isn't political, just poor English. There is no need to personify the user, as it is a concept.
If for example, (without that) "If anyone wants to use the fire escape in case of a fire, they will find it around the corner." "they" can be employed.

File: frontend/apps/reader/modules/readergesture.lua (hunk 3 above)
Comment (comradekingu) [inline frontend/apps/reader/modules/readergesture.lua]:
```suggestion
            logger.info("UI language direction changed: Resetting some gestures to directional defaults.")
```

Comment (comradekingu) [reply frontend/apps/reader/modules/readergesture.lua]: Pretty sure I saw both. Using the one that is only allowed in English baits errors in translation.

File: frontend/apps/reader/modules/readergesture.lua (hunk 4 above)
Comment (comradekingu) [inline frontend/apps/reader/modules/readergesture.lua]:
```suggestion
        new_text = _("Accelerometer rotation events off.")
```

File: frontend/apps/reader/modules/readergesture.lua [hunk 5]
```diff
@@ -1601,9 +1601,9 @@ end
...
     else
-        new_text = _("Accelerometer rotation events will now be honored.")
+        new_text = _("Accelerometer rotation events off.")
```
Comment (comradekingu) [inline frontend/apps/reader/modules/readergesture.lua]:
```suggestion
        new_text = _("Accelerometer rotation events on.")
```

File: frontend/apps/reader/modules/readergesture.lua (hunk 1 above)
Comment (comradekingu) [inline frontend/apps/reader/modules/readergesture.lua]:
```suggestion
    wifi_off = _("Turn off Wi-Fi"),
```

File: frontend/apps/reader/modules/readergesture.lua (hunk 3 above)
Comment (Frenzie) [reply frontend/apps/reader/modules/readergesture.lua]: Could you please point out the problematic strings? This is just random code.

File: frontend/apps/reader/modules/readergesture.lua (hunk 2 above)
Comment (poire-z) [reply frontend/apps/reader/modules/readergesture.lua]: If we can't use poor english in comments, I'm gonna stop putting comments :|

(But I doubt it's worse than "it" or "they"... I often use "the user" when I refer to users-only in comments - when I write about how developpers would/should do things, I write "you". I respect both of them, and neither is more "a concept" than the others ! They're both human beings :)

And at least, "him", personifying the user, drop some ambiguity:
What would be "it" in your initial suggestion? the user? the code that follows? the widget class we're implementing? the widget object that will be instantiated?
With "him", no ambiguity.

So please, leave this comment as it is - and ignore the above suggestion as it is dropping some contextual logic by removing the leading `If not, `. Unclear logic is worse than poor english.

File: frontend/apps/reader/modules/readergesture.lua (hunk 3 above)
Comment (poire-z) [reply frontend/apps/reader/modules/readergesture.lua]: Indeed, that's just some string that is logged in crash.log - it shouldn't reach weblate in any way - or does it ?!

Comment (poire-z) [reply frontend/apps/reader/modules/readergesture.lua]: I don't like this one either :) even if directional is the english adjective for direction, I'd rather keep the word direction to again be explicite (as we're not all native english speakers, simple english is sometimes better in comments).
But it's just a log, and if you'd really want it to be clear and unambiguous, you'd use:
```diff
-UI language direction changed: resetting some gestures to direction default
+UI language direction changed: Resetting some gestures to their default for the new direction.
```

File: frontend/apps/reader/modules/readergesture.lua (hunk 2 above)
Comment (Frenzie) [reply frontend/apps/reader/modules/readergesture.lua]: Using "he" is perhaps slightly outdated stylistically, but it's definitely not poor English.

File: frontend/apps/reader/modules/readergesture.lua (hunk 3 above)
Comment (Frenzie) [reply frontend/apps/reader/modules/readergesture.lua]: The original sentence would be better as "direction defaults."

As an aside, for comments & logs I purposefully try not suggest corrections or improvements as long as they're comprehensible and not ungrammatical. Because there the goal is the future maintainability of the code.

**Comment by pazos** [comment]
In spanish, galician etc. "Apaga/enciende el wifi" sounds horrible and is not used outside the context of physical turning on/off wifi routers. What users will do is "comprobar si el wifi esta activado" in their phones.

Both MacOS, Linux and Android have actions to "Desactivar el WiFi", not for "Apagar el Wifi". On some devices, like the Cervantes, is plain false as wireless chip cannot be turned off.

In short: enable/disable is what we do. In some devices it does imply turning on/off vcore but that's too technical to be presented to the user. Disable means: drop a lease, break a link, unload a module. Enable the oposite: load a module, link to ap, get a lease.

**Comment by Frenzie** [comment]
> In spanish, galician etc.

That's not relevant to English I'm afraid. That means the proper translation of "turn off" would be "desactivar." No more, no less. :-)

> In short: enable/disable is what we do. In some devices it does imply turning on/off vcore but that's too technical to be presented to the user. 

By this argument, the conclusion may well be the reverse. Turn on/off is how regular people talk, while enable/disable is more technical.

**Comment by comradekingu** [comment]
@Frenzie Could just the name of the thing being either on/off/activated/deactivated be used, with a checkbox next to it?

File: frontend/apps/reader/modules/readergesture.lua (hunk 1 above)
Comment (Frenzie) [reply frontend/apps/reader/modules/readergesture.lua]: No one said anything about the claims of the Wi-Fi Alliance. What I wrote is that no matter if you look at it as an abbreviation of wireless fidelity or as a generic descriptor, the correct spelling is _wifi_.

WLAN is something like Central European, not exactly regular English. It can be an excellent translation for wifi/Wi-Fi. Wireless network is a bit verbose and possibly goofy, but better.

As for any "pollution," such loaded language makes me want to take up a principled stance against Wi-Fi pollution. :imp: But seriously, a translation memory can't be "polluted" by a slightly different spelling. Next thing you know someone will spell _color_ like _colour_. Oh, the humanity…

**Comment by Frenzie** [comment]
@comradekingu That's the general pattern. Turn on wifi/W-Fi is one of a small number of exceptions because it occurs in a scenario like:

1. The user selects an action that requires network access.
2. The program notices a lack of networking, so it pops up a dialog.

**Comment by comradekingu** [comment]
> Mhhh, these changes feel all gratuitous, less precise, or wrong. (I seem to prefer our current enable/disable vs turn on/off - but english being not my first language...)

You are turning off functionality. The distinction is when capability is underlying said functionality, but I don't think that is the case here (?)

File: frontend/apps/reader/modules/readergesture.lua (hunk 1 above)
Comment (comradekingu) [reply frontend/apps/reader/modules/readergesture.lua]: I grant you that point, and it is one of semantics. Apply it to that of "Wi-Fi*", and you will find nobody spells it _wifi".

https://dictionary.cambridge.org/dictionary/english/wi-fi?q=wifi
https://www.dictionary.com/browse/wifi?s=t
https://dictionary.cambridge.org/dictionary/english/wi-fi?q=wifi
https://www.vocabulary.com/dictionary/WiFi https://en.wikipedia.org/wiki/Wi-Fi

Meanwhile, spelling it "Wi-Fi", _isn't wrong_, _and_ enjoys the most popularity.

Some readers have 3G/WiMAX/4G and so on too. My personal preference is to use something more generic when it is possible. Even https://emojipedia.org/antenna-with-bars/ would do.

**Comment by pazos** [comment]
> That's not relevant to English I'm afraid. That means the proper translation of "turn off" would be "desactivar." No more, no less. :-)

Exactly. Translations have their own life, I guess :+1:

File: frontend/apps/reader/modules/readergesture.lua (hunk 2 above)
Comment (NiLuJe) [reply frontend/apps/reader/modules/readergesture.lua]: @poire-z: I *really* wouldn't worry about that ;). We're being nitpicky *because* the PR itself is nitpicky. Kind of a vicious circle ;).

File: frontend/apps/reader/modules/readergesture.lua (hunk 1 above)
Comment (Frenzie) [reply frontend/apps/reader/modules/readergesture.lua]: Nobody except for Nerdist, Business Insider, Gizmodo, Techcrunch, Public Contract Law Journal, Analog, Literary Review, Daily Beast, MotherJones, NewStatesMan, Denver, CSMonitor, and FantasySciFi, just to name a few. Many other publications use WiFi, Wifi, as well as wi-fi. TIME Magazine switched from Wi-Fi to wi-fi in 2003 when it turned generic.

Note that I [said](https://github.com/koreader/koreader/pull/5746#discussion_r363202383) I want to be part of the change:

> I recognize that in well-edited English prose, wifi is still in a minority. This will change.

I assume you included the Cambridge Dictionary twice to emphasize that it does, in fact, fully agree with me. ;-) The fact that it happens to spell the words in question _hi-fi_ and _wi-fi_ is not the salient point, only that it treats them the same.

PS I thought the corpus of US Supreme Court Opinions was quite amusing.

https://caselaw.findlaw.com/us-supreme-court/13-896.html
> Suppose an extensive business headquarters or a resort or a college campus wants a single, central wireless system (sometimes called a Wi-Fi network). In order to cover the large space, the system needs multiple base stations so a user can move around the area and still stay connected.

**Comment by Frenzie** [comment]
> Edit: "Turn on" should be "OK" there. (According to convention.)

Convention in the '90s perhaps. ;-) This design pattern has been well established in the 2000s by Apple and GNOME.

https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/dialogs-and-flyouts/dialogs
dialog_rs2_delete_file

https://developer.apple.com/design/human-interface-guidelines/macos/windows-and-views/dialogs/
Dialog-buttons_2x

https://material.io/components/dialogs/#usage
mio-components_assets_1Vs3vgE4jR4V5x-jjaQUhC1LqdOmdi2o2_dialogs-alertdialog

https://developer.gnome.org/hig/stable/dialogs.html.en
Screenshot_2020-01-06_19-01-02

**Comment by Frenzie** [comment]
Anyway, let's just drop the changes to the comments & keep the rest of the PR as is (with turn on/off and Wi-Fi).

File: frontend/apps/reader/modules/readergesture.lua (hunk 2 above)
Comment (poire-z) [reply frontend/apps/reader/modules/readergesture.lua]: Thanks :) but I don't _really_ worry - if we can't put a bit of drama-dy in our GitHub comments, I'm gonna stop writting GitHub comments! :)
Just trying to discourage rich-english translators from too easily thinking they can change comments or non-translated strings for pseudo grammar reasons, without having to read and understand the code and logic surrounding these comments (because the code _is_ some comment to the comments - in case the comments are not clear enough :) - better they not contradict each others).

File: frontend/apps/reader/modules/readergesture.lua (hunk 3 above)
Comment (Frenzie) [reply frontend/apps/reader/modules/readergesture.lua]: Using the comment suggest functionality to fix this up for merge.
```suggestion
            logger.info("UI language direction changed: resetting some gestures to their default for the new direction.")
```

Comment (Frenzie) [reply frontend/apps/reader/modules/readergesture.lua]: @comradekingu You didn't enable push access. Could you please fix up/revert the random comments so the PR can be merged?

Comment (comradekingu) [reply frontend/apps/reader/modules/readergesture.lua]: @Frenzie I did, but your are confusing the plural "gestures" with the singular "default". I didn't randomly change that.

**Comment by comradekingu** [comment]
@Frenzie Not sure what discussion this is. There is _plenty_ of stuff to actually fix in KOReader.
I think the greater in this specific case is to explain what is happening. Adding the imperative up front gets to the point quicker, with less arbitration.
"Turn on Wi-Fi to *?"
Cancel | OK

File: frontend/apps/reader/modules/readergesture.lua (hunk 1 above)
Comment (comradekingu) [reply frontend/apps/reader/modules/readergesture.lua]: @Frenzie You will never see anything but "Wi-Fi" from the Wi-Fi alliance. This ensures anything but "Wi-Fi", which is in clear minority for good reason, will always be and look like a knock-off.
My changes improve consistency within KOReader, and also help avoid errors in translation, as anything but "Wi-Fi" isn't allowed elsewhere.
Using "wifi" for wireless network is an erroneous practice.

File: frontend/apps/reader/modules/readergesture.lua (hunk 2 above)
Comment (comradekingu) [reply frontend/apps/reader/modules/readergesture.lua]: So you use "he", for archaic reasons, sure. The odds of anyone thinking that isn't in poor taste and for no good reason is slim. Slimmer still considering the state of the rest of the strings in KOReader. Speaking of users, lets have them, and lets consider what they are likely to understand rather than what is merely a technically dependable position.

File: frontend/apps/reader/modules/readergesture.lua (hunk 3 above)
Comment (comradekingu) [reply frontend/apps/reader/modules/readergesture.lua]: @Frenzie Didn't find any, but most are unexplained placeholders, so could be I remembered it as being one of those. https://github.com/koreader/koreader/blob/a7def7e8338659e51f4bc956487787391b76a41a/frontend/apps/reader/modules/readerwikipedia.lua#L484 is off, and I saw about 10 other clear errors looking for it.
Do you want me to find evidence of errors in translation resulting from : small-versal?

**Comment by Frenzie** [comment]
@comradekingu The discussion was about passive labels.

**Comment by pazos** [comment]
@comradekingu: please differenciate between issues in the Ui and issues in the translation. You are making translation changes in this PR. If you want to make Ui changes then please open other PR. If you don't want to make the changes yourself but suggest them for others to do the work, then open a new ticket per each issue describing current behaviour and expected behaviour.

That makes things easier for contributions.

File: frontend/apps/reader/modules/readergesture.lua (hunk 2 above)
Comment (NiLuJe) [reply frontend/apps/reader/modules/readergesture.lua]: Doesn't change the fact that `it` is ungrammatical in this context?

(TL;DR: While "he" isn't great, at least it makes a proper sentence).

File: frontend/apps/reader/modules/readergesture.lua (hunk 3 above)
Comment (Frenzie) [reply frontend/apps/reader/modules/readergesture.lua]: @comradekingu 
Screenshot_2020-02-12_18-36-25

> but you are confusing the plural "gestures" with the singular "default".

You can certainly say English generally prefers a plural noun for a repeated idea, but that's besides the point.

Comment (NiLuJe) [reply frontend/apps/reader/modules/readergesture.lua]: Except it's "value" taking the plural in his suggestion. default values OR defaults. defaults values would be wrong, though.

**Comment by Frenzie** [comment]
On that note, this PR has been blocked for over a month because it changes more than what it says on the tin (Wi-Fi and turn on/off). Ideally those would be three separate PRs.

One for Wi-Fi (probably merged), one for turn on/off (merged immediately), and one for nitpicking internal comments (probably not merged but in any case not holding up anything else).

File: frontend/apps/reader/modules/readergesture.lua (hunk 3 above)
Comment (Frenzie) [reply frontend/apps/reader/modules/readergesture.lua]: @NiLuJe I just edited it to "default values"; it just said "default" previously because I copied it straight from @poire-z as a statement. ;-)

I figured most people would miss the subtle statement; I wasn't quite expecting it to be caught yet still missed. ^_^

Comment (NiLuJe) [reply frontend/apps/reader/modules/readergesture.lua]: @Frenzie: Ah, okay, now it makes sense ;). Thanks.

Comment (Frenzie) [reply frontend/apps/reader/modules/readergesture.lua]: Yes please! :-)

File: frontend/apps/reader/modules/readergesture.lua (hunk 1 above)
Comment (Frenzie) [reply frontend/apps/reader/modules/readergesture.lua]: @comradekingu https://github.com/koreader/koreader/pull/5746#issuecomment-571270197

> Anyway, let's just drop the changes to the comments & keep the rest of the PR as is (with turn on/off and Wi-Fi).

**Comment by comradekingu** [comment]
@Frenzie Ideally I would have the time, but I don't. KOReader has 10k words that I haven't gotten through yet. If at some time it stops being all over the place, one could start going easier on the changes. For how long it has been this way, it seems me doing something about it is how that would end up getting done.

**Comment by Frenzie** [comment]
Changes merged in #5856.
